wcfsetup/install/files/lib/util/StringUtil.class.php

```diff
@@ -119,20 +119,8 @@ public static function encodeHTML($string) {
 	 * @return	string
 	 */
 	public static function encodeJS($string) {
-		// unify newlines
 		$string = self::unifyNewlines($string);
-		
-		// escape backslash
-		$string = str_replace("\\", "\\\\", $string);
-		
-		// escape singe quote
-		$string = str_replace("'", "\'", $string);
-		
-		// escape new lines
-		$string = str_replace("\n", '\n', $string);
-		
-		// escape slashes
-		$string = str_replace("/", '\/', $string);
+		$string = str_replace(["\\", "'", "\n", "/"], ["\\\\", "\'", '\n', '\/'], $string);
```

Comment:
Consider: [`addcslashes($string, "\\'\n/")`](http://php.net/manual/en/function.addcslashes.php).

Reply:
Does it actually make any difference? `str_replace()` is both binary safe and does not care for the null byte, unlike `addcslashes()`.

Reply:
I just tried: For a string with 10000 bytes in length `str_replace` is faster. For 100000 bytes they are about the same.

You'd expect that the specialized function is the faster one …

So: Disregard that comment.